js/lib/bands.js

```diff
@@ -192,6 +206,28 @@ function bandPlot(bandDivId, bandPathTextBoxId, dataFilePaths, dosFile, showFerm
             };
             a.click();
         }
+
+        var theDownloadJsonButton = document.getElementById(bandDivId + "bt-downloadJson");
+        theDownloadJsonButton.onclick = function () {
+            var a = document.createElement('a');
+
+            if (dataFilePaths.length) {
```

Comment:
I think the `length` is to check if the data exist. So rather than check the length it would be clear to check the existence directly.

Reply:
dataFilePaths can be empty, if only DOS file was given.

Reply:
Okay, I see, but the for dos and bands data here the variable's name are messed up and very confusion. If you can fix it in another PR, that would be helpful for future maintenance. Can you do that?

Reply:
I agree this vriable is confusion. I am going to change it.